awslabs/aws-iot-device-client
# Provisioning with Secure Element feature release.

### Motivation
- Storing secret key on a device file system is not secure. It is a security best practice to store keys in a more secure physical location, which does not allow any unauthorized person to access or make copies of the key. AWS IoT Secure Element feature allows user to use private key stored in PKCS#11 compatible smart cards or Hardware Security Module (HSM).


### Modifications
#### Change summary
Please describe what changes are included in this pull request. 
* Updated DC config to read, validate and store feature config
* Updated SharedCrtResourceManager to connect to IoT core using PKCS11 object
* Added supporting documentation for the feature change


#### Revision diff summary
If there is more than one revision, please explain what has been changed since the last revision.

### Testing
 **Is your change tested? If not, please justify the reason.**  
 **Please list your testing steps and test results.** 
- Manually tested feature change
- Added unit tests for the change


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
